tests/state/test_state_api.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from remoulade.api.main import app
+from remoulade.state import StateNamesEnum
+
+
+@pytest.fixture
+def api_client(state_middleware):
+    with app.test_client() as client:
+        yield client
+
+
+class TestMessageStateAPI:
+    """ Class Responsible to do the test of the API """
+
+    def test_no_messages(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/messages/states/")
+        assert res.status_code == 200
+        assert res.data == b"[]\n"
+
+    def test_invalid_url(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/invalid_url/")
+        assert res.status_code == 404
+
+    def test_pending_message(self, stub_broker, state_middleware, do_work, api_client):
+        do_work.send()
+        res = api_client.get("/messages/states/")
+        for state_enum in StateNamesEnum:
```

Comment:
you should not have to do these for loop to get an enum. It's because of (in states.enum)
```
    def __get__(self, instance, owner):
        return self.value
```
We should not overide how enum works.

Sorry, I thougth it was good but after som though it's not.

I made some changes to the enums:
```
class StateNamesEnum(Enum):
    """
    Contains the possible states that can have a message:
    """

    Started = "Started"  # a `Message` that has not been processed
    Pending = "Pending"  # a `Message` that has been enqueued
    Skipped = "Skipped"  # a `Message` that has been skipped
    Canceled = "Canceled"  # a `Message` that has been canceled
    Failure = "Failure"  # a `Message` that has been processed and raised an Exception
    Success = "Success"  # a `Message` that has been processed and does not raise an Exception


#: A type alias representing states in the database
class State(namedtuple("State", ("message_id", "name", "args", "kwargs"))):
    """Catalog Class, it storages the state
        Parameters:
            name: Name of the state
            args: List of arguments in the state(name)
    """

    def __new__(cls, message_id, name, args, kwargs):
        if name not in [state for state in StateNamesEnum]:
            raise InvalidStateError("The {} State is not defined".format(name))
        return super().__new__(cls, message_id, name, args, kwargs)

    @classmethod
    def from_dict(cls, dict):
        dict["name"] = StateNamesEnum(dict["name"])
        return cls(**dict)

    def asdict(self):
        return {**self._asdict(), "name": self.name.value}
```

This way you will be able to do:
`assert StateNamesEnum.Pending.name.encode() in res.data`